### PR TITLE
fix: critical undefined balance error

### PR DESCRIPTION
**Pull request by alter-eggo**
This pr fixes critical undefined balance err
image_480

### CODE REVIEW

File: lib/modules/pool/queries/useOnchainUserPoolBalances.ts [hunk 1]
```diff
@@ -111,11 +111,19 @@ function overwriteOnchainPoolBalanceData(
 
     // Unstaked balances
     const onchainUnstakedBalances = ocUnstakedBalances[pool.id]
+    if (!onchainUnstakedBalances) {
+      return pool
+    }
```
Comment (garethfuller) [inline lib/modules/pool/queries/useOnchainUserPoolBalances.ts:116]: What if they have staked balances but not unstaked? You are returning early here but that means you're skipping over potential `onchainStakedBalances` right?

Comment (garethfuller) [reply lib/modules/pool/queries/useOnchainUserPoolBalances.ts:116]: Ok, so lets handle that properly. Please catch this in Sentry.

Comment (alter-eggo) [reply lib/modules/pool/queries/useOnchainUserPoolBalances.ts:116]: they are already there https://github.com/balancer/frontend-v3/blob/main/lib/modules/pool/queries/useOnchainUserPoolBalances.ts#L49-L55

Comment (garethfuller) [reply lib/modules/pool/queries/useOnchainUserPoolBalances.ts:116]: I don't think they are capturing these issues, are they? Where's an example in Sentry?

**Comment by garethfuller** [comment]
I noticed this on production and it causes a complete app crash, so merging just to prevent users seeing that.

**Pull request merged**